src/util/retrieveNoteReferredEntityDataFromLocationState.js

```diff
@@ -0,0 +1,13 @@
+const retrieveNoteReferredEntityDataFromLocationState = (state) => {
```

Comment:
Complex naming. Call it simpler, like `getReferredEntityData`, and you can add a function description if you like.